Objects/bytearrayobject.c

```diff
@@ -335,9 +335,18 @@ bytearray_repeat(PyByteArrayObject *self, Py_ssize_t count)
         if (mysize == 1)
             memset(result->ob_bytes, buf[0], size);
         else {
-            Py_ssize_t i;
-            for (i = 0; i < count; i++)
-                memcpy(result->ob_bytes + i*mysize, buf, mysize);
+            Py_ssize_t i, j;
+
+            i = 0;
+            if (i < size) {
+                memcpy(result->ob_bytes, buf, mysize);
+                i = mysize;
+            }
+            while (i < size) {
+                j = (i <= size - i) ? i : size - i;
```

Comment:
I'd suggest using `Py_MIN(i, size-i)` here. Also, consider adding a comment that this is repeatedly doubling the bytes.